### DISCUSSION TITLE
difference between admin api & dashboard

### DISCUSSION THREAD

**Discussion by caibirdme** [apache/apisix#6180] [Q&A] [answered]
Hi, I want to know the difference between apisix admin api and apisix-dashboard. If I need to add route dynamically, which way should I choose? Calling admin API directly or calling dashboard api？

**Comment by starsz** [comment]
Hi, @caibirdme 
Both are OK.
But better to use Apache APISIX admin API. Since its doc is more rich and stable.
The doc of Admin API : https://apisix.apache.org/docs/apisix/admin-api

The Apache APISIX Dashboard is designed to make it as easy as possible for users to operate Apache APISIX through a frontend interface : refer: https://github.com/apache/apisix-dashboard#whats-apache-apisix-dashboard

**Reply by caibirdme** [reply]
Thx, and I want to know is there any available SDK that I can use?

**Reply by starsz** [reply]
Maybe you can refer the apisix client in [Apache APISIX-Ingress-Controller](https://github.com/apache/apisix-ingress-controller).

Refer: https://github.com/apache/apisix-ingress-controller/blob/master/pkg/apisix/apisix.go

**Comment by leslie-tsang** [comment, marked as answer]
Hello there, `Admin API` is the build-in **REST API** of APISIX, Normally, It's recommended to use `Admin API` instead of `apisix-dashboard` according to your scenario. Both connect directly to the etcd, then distribute the change to the APISIX cluster.

> I want to know the difference between apisix admin api and apisix-dashboard

Admin API do the jsonschema check via a plugin-defined schema, `apisix-dashboard` do the jsonschema check with [schema.json](https://github.com/apache/apisix-dashboard/blob/master/api/conf/schema.json)

> Thx, and I want to know is there any available SDK that I can use?
 
Not supported yet.
